### PR TITLE
Users/rok/telemetry unique vm final

### PR DESCRIPTION
**Pull request by rokmicrosoft**

### CODE REVIEW

File: Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1 [hunk 1]
```diff
@@ -29,6 +29,26 @@ Import-Module "$PSScriptRoot\DeploymentUtilities\Microsoft.TeamFoundation.Distri
...
+       "," , 
+       "`"VmUuidHash`" : `"$($deploymentResponse.VmUuidHash)`"" , 
+       "," , 
+       "`"TelemetryError`" : `"$($deploymentResponse.TelemetryError)`"" ,
+       "," ,
+       "`"JobId`" : `"$jobId`"" ,
+       "}")
+    }
+
+    $telemetryString ="##vso[telemetry.publish area=TaskHub;feature=PowerShellOnTargetMachines]$jsonString"
```
Comment (kmkumaran) [inline Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1]: @RoopeshNair - let us know if you are ok with this telemetry data.

File: Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1 [hunk 2]
```diff
@@ -103,7 +123,7 @@ try
             $status = $deploymentResponse.Status
 
             Write-Output (Get-VstsLocString -Key "PS_TM_DeploymentStatusForMachine01" -ArgumentList $displayName, $status)
-
+            Publish-Azure-Telemetry  -deploymentResponse $deploymentResponse -jobId $jobId
```
Comment (chshrikh) [inline Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1]: extra space after Publish-Azure-Telemetry

File: Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1 [hunk 3]
```diff
@@ -29,6 +29,26 @@ Import-Module "$PSScriptRoot\DeploymentUtilities\Microsoft.TeamFoundation.Distri
...
+function Publish-Azure-Telemetry
+ {
+   param([object] $deploymentResponse, [string] $jobId)
+    if($deploymentResponse){
+       $jsonString = -join("{" , 
+       "`"IsAzureVm`" : `"$($deploymentResponse.IsAzureVm)`"" , 
+       "," , 
+       "`"VmUuidHash`" : `"$($deploymentResponse.VmUuidHash)`"" , 
+       "," , 
+       "`"TelemetryError`" : `"$($deploymentResponse.TelemetryError)`"" ,
```
Comment (chshrikh) [inline Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1]: assuming the test cases where  TelemetryError is null and not null both are working fine

File: Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1 [hunk 4]
```diff
@@ -29,6 +29,26 @@ Import-Module "$PSScriptRoot\DeploymentUtilities\Microsoft.TeamFoundation.Distri
 # Telemetry
```
Comment (chshrikh) [inline Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1]: ensure you hold this PR till you update the DeploymentUtilities.zip , else the empty values will get publish as part of Publish-Azure-Telemetry

File: Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1 (hunk 2 above)
Comment (rokmicrosoft) [reply Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1]: removed extra space

File: Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1 (hunk 3 above)
Comment (rokmicrosoft) [reply Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1]: Done.

File: Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1 (hunk 4 above)
Comment (rokmicrosoft) [reply Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1]: Updated DeploymentUtilities.zip url and patch version

File: Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1 (hunk 1 above)
Comment (rokmicrosoft) [reply Tasks/PowerShellOnTargetMachines/PowerShellOnTargetMachines.ps1]: Discussed in mail.

File: Tasks/PowerShellOnTargetMachines/task.json [hunk 5]
```diff
@@ -13,7 +13,7 @@
     "version": {
         "Major": 2,
         "Minor": 0,
-        "Patch": 2
+        "Patch": 3
```
Comment (chshrikh) [inline Tasks/PowerShellOnTargetMachines/task.json]: update this with loc json as well

Comment (rokmicrosoft) [reply Tasks/PowerShellOnTargetMachines/task.json]: Done.

**Pull request merged**